# Exercise 4.6_002

In [2]:
# import libraries
import pandas as pd
import numpy as np
import os

In [3]:
path = r'C:\Users\Tim\Desktop\Data Analytics Intro\Python Fundamentals for Data Analysts 4\Instacart Basket Analysis'

In [4]:
path

'C:\\Users\\Tim\\Desktop\\Data Analytics Intro\\Python Fundamentals for Data Analysts 4\\Instacart Basket Analysis'

In [17]:
df_ords_prods_combined = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_combined.pkl'))

In [18]:
df_ords_prods_combined.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2539329        1             1          2                  8   
2   2539329        1             1          2                  8   
3   2539329        1             1          2                  8   
4   2539329        1             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  
0                     NaN         196                  1          0   both  
1                     NaN       14084                  2          0   both  
2                     NaN       12427                  3          0   both  
3                     NaN       26088                  4          0   both  
4                     NaN       26405                  5          0   both

In [19]:
df_ords_prods_combined.shape

(32434489, 10)

In [38]:
df_prods_checked = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'products_data_checked.csv'), index_col = False)

In [39]:
df_prods_checked.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [41]:
# there is a need to drop the Unnamed column becuase it is not useful
df_prods_checked = df_prods_checked.drop(['Unnamed: 0'], axis=1)

In [42]:
df_prods_checked.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [43]:
df_prods_checked.shape

(49693, 5)

# next is to drop column that is not relevant

In [ ]:
df_ords_prods_combined = df_ords_prods_combined.drop(['_merge'], axis=1)

In [32]:
# check output
df_ords_prods_combined.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2539329        1             1          2                  8   
2   2539329        1             1          2                  8   
3   2539329        1             1          2                  8   
4   2539329        1             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered  
0                     NaN         196                  1          0  
1                     NaN       14084                  2          0  
2                     NaN       12427                  3          0  
3                     NaN       26088                  4          0  
4                     NaN       26405                  5          0

In [33]:
df_ords_prods_combined.shape

(32434489, 9)

In [45]:
df_ords_prods_merge = df_prods_checked.merge(df_ords_prods_combined, on = 'product_id', indicator = True)

In [46]:
df_ords_prods_merge.head()

product_id                product_name  aisle_id  department_id  prices  \
0           1  Chocolate Sandwich Cookies        61             19     5.8   
1           1  Chocolate Sandwich Cookies        61             19     5.8   
2           1  Chocolate Sandwich Cookies        61             19     5.8   
3           1  Chocolate Sandwich Cookies        61             19     5.8   
4           1  Chocolate Sandwich Cookies        61             19     5.8   

   order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   3139998      138            28          6                 11   
1   1977647      138            30          6                 17   
2    389851      709             2          0                 21   
3    652770      764             1          3                 13   
4   1813452      764             3          4                 17   

   days_since_prior_order  add_to_cart_order  reordered _merge  
0                     3.0                  5          0   both  
1                    20.0                  1          1   both  
2                     6.0                 20          0   both  
3                     NaN                 10          0   both  
4                     9.0                 11          1   both

In [36]:
df_ords_prods_merge.shape

(32434212, 14)

# Confirm the results of the merge using the merge flag

In [37]:
df_ords_prods_merge['_merge'].value_counts()

_merge
both          32434212
left_only            0
right_only           0
Name: count, dtype: int64

After the merge, the dataframe now has total of 32434212 rows. It all have combination of information that are used for the entire project. 

# Export the newly created dataframe as ords_prods_merge

In [47]:
# Export to pkl
df_ords_prods_merge.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_merge.pkl'))